src/components/patch-details/patch-details.tsx

```diff
@@ -17,16 +32,114 @@ interface PatchDetailsProps {
   const { heading } = locale.patchDetails;
 
   const { patchLabel, housingOfficerLabel, areaManagerLabel } = locale.patchDetails;
 
   const patchOrAreaDefined = assetPatch || assetArea;
   const housingOfficerName = assetPatch?.responsibleEntities[0]?.name;
   const areaManagerName = assetArea?.responsibleEntities[0]?.name;
+  const [newPatchName, setNewPatchName] = useState<string>();
+  const [isEditingPatchName, setIsEditingPatchName] = useState(false);
+  const [patchesAndAreas, setPatchesAndAreas] = useState<Patch[]>([]);
+
+  useEffect(() => {
+    getAllPatchesAndAreas().then((data) => {
+      setPatchesAndAreas(
+        data.filter(
+          (patchOrArea) =>
+            ![
+              "Hackney",
+              "CL Area",
+              "CP Area",
+              "HN1 Area",
+              "HN2 Area",
+              "SD Area",
+              "SH Area",
+              "SN Area",
+            ].includes(patchOrArea.name),
+        ),
+      );
+    });
+  }, []);
+
+  const updatePropertyPatchCall = (request: UpdatePropertyPatchRequest) => {
+    updatePropertyPatch(assetPk, request, versionNumber?.toString() ?? "")
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        window.location.reload();
+      });
+  };
+  const getbyPatchNameCall = (patchName: string) => {
+    getByPatchName(patchName)
+      .then((data) => {
+        const id = JSON.stringify(data).match(/"id":"(.*?)"/);
+        const parentId = JSON.stringify(data).match(/"parentId":"(.*?)"/);
+        console.log("id L84", id);
+        console.log("parentId L85", parentId);
+        if (!id?.[1] || !parentId?.[1]) {
+          throw new Error("Invalid patch data: id or parentId is undefined");
+        }
+        const request = buildUpdatePropertyPatchRequest(id[1], parentId[1]);
+        updatePropertyPatchCall(request);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  };
+  const handleEdit = () => {
+    console.log("newPatchName", newPatchName);
+    if (newPatchName) {
+      getbyPatchNameCall(newPatchName);
+    }
+  };
+
+  function SetPatchNameOption() {
+    return (
+      <div className="patch" key="patchName">
+        <select
+          id="patch-dropdown-options"
+          className="govuk-input lbh-input"
+          data-testid="patch-dropdown-options"
+          value={newPatchName}
+          onChange={(e) => setNewPatchName(e.target.value)}
+        >
+          <option disabled value="">
+            {" "}
+            -- Select an option --{" "}
+          </option>
+          <option selected>{assetPatch?.name}</option>
+          {patchesAndAreas
+            .filter((patchOrArea) => patchOrArea.patchType === "patch")
+            .sort((a, b) => (a.name > b.name ? 1 : -1))
+            .map(({ name }) => (
+              <option key={name} value={name}>
+                {name}
+              </option>
+            ))}
+        </select>
+      </div>
+    );
+  }
```

Comment:
Shouldn't this be its own component? Adding a component within a component isn't ideal because 1. It can be bad for performance by interfering with the react lifecycle. 2. It makes this difficult to refactor because it relies on global variables